From 0ddcd8785199c08d9bded3c767a3b1227c670999 Mon Sep 17 00:00:00 2001
From: Joaquim Monteiro <joaquim.monteiro@protonmail.com>
Date: Sun, 23 Jun 2024 21:17:25 +0100
Subject: [PATCH 1/2] os: Fix assignment with incompatible pointer type

struct hostent->h_addr_list is of type char**, not const char**.
GCC considers this an error when in C99 mode or later.

Signed-off-by: Joaquim Monteiro <joaquim.monteiro@protonmail.com>
Part-of: <https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1572>
---
 os/access.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os/access.c b/os/access.c
index 7d5e3b9112..bb25aac30e 100644
--- a/os/access.c
+++ b/os/access.c
@@ -1842,7 +1842,7 @@ siHostnameAddrMatch(int family, void *addr, int len,
         char hostname[SI_HOSTNAME_MAXLEN];
         int f, hostaddrlen;
         void *hostaddr;
-        const char **addrlist;
+        char **addrlist;
 
         if (siAddrLen >= sizeof(hostname))
             return FALSE;
-- 
GitLab


From a6a993f9508ff0bd6f19695b2dfaef170c652525 Mon Sep 17 00:00:00 2001
From: Joaquim Monteiro <joaquim.monteiro@protonmail.com>
Date: Sun, 23 Jun 2024 21:23:47 +0100
Subject: [PATCH 2/2] os: Fix siHostnameAddrMatch in the case where h_addr
 isn't defined

When IPv6 support isn't enabled, and h_addr isn't defined,
there is no for loop, so the break statement is invalid.

Signed-off-by: Joaquim Monteiro <joaquim.monteiro@protonmail.com>
Part-of: <https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1572>
---
 os/access.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os/access.c b/os/access.c
index bb25aac30e..8eadb3edaa 100644
--- a/os/access.c
+++ b/os/access.c
@@ -1867,7 +1867,9 @@ siHostnameAddrMatch(int family, void *addr, int len,
                 if ((f == family) && (len == hostaddrlen) &&
                     (memcmp(addr, hostaddr, len) == 0)) {
                     res = TRUE;
+#ifdef h_addr
                     break;
+#endif
                 }
             }
         }
-- 
GitLab

